### PR TITLE
Add support for --generate-self-signed-cert

### PR DESCRIPTION
**Pull request by fantix**
This PR includes:

* New TLS options in the command line: `--generate-self-signed-cert`, `-tls-cert-file` and `--tls-key-file`
* New Docker environment variables: `EDGEDB_GENERATE_SELF_SIGNED_CERT`, `EDGEDB_CERT_FILE` and `EDGEDB_KEY_FILE`
* Friendly error if none of the above is provided and no existing certificate file is found
* Echo back the generated certificate for Docker users (hidden for CLI users)

### CODE REVIEW

**Review by elprans — COMMENTED** [review]
We also need to add support for `CERT_FILE` and `KEY_FILE` too.

**Comment by fantix** [comment]
> We also need to add support for CERT_FILE and KEY_FILE too.

I'll fix that in a separate PR.

File: README.md [hunk 1]
```diff
@@ -110,6 +121,22 @@ verifier instead of plain text.
...
+
+Set this option to `1` to tell the server to automatically generate a
+self-signed certificate with key file in the `EDGEDB_DATADIR` (if present,
+see below), and echo the certificate content in the logs. If the certificate
+file exists, the server will use it instead of generating a new one.
+
+Self-signed certificates are usually used in development and testing, you
+should likely provide your own certificate and key file with the variables below.
+
+#### `EDGEDB_CERT_FILE`, `EDGEDB_KEY_FILE`
```
Comment (elprans) [inline README.md:134]: Let's use EDGEDB_TLS_CERT_FILE and EDGEDB_TLS_KEY_FILE for consistency with command line args

Comment (fantix) [reply README.md:134]: Sure thing!

**Review by elprans — APPROVED** [review]

**Pull request merged**